pkg/traceql/combine.go

```diff
@@ -70,6 +70,17 @@ func combineSearchResults(existing *tempopb.TraceSearchMetadata, incoming *tempo
 		existing.DurationMs = incoming.DurationMs
 	}
 
+	// Merge service stats
+	for service, incomingStats := range incoming.ServiceStats {
+		existingStats, ok := existing.ServiceStats[service]
```

Comment:
i don't think this merge works. if we find multiple trace fragments that satisfy a traceql result we will be repeatedly re-adding the service span/error counts.

should it be max?

Reply:
Good point, I think max is more correct

Reply:
If you both agree on using `max()`, I've created a PR targeting the branch of this PR: https://github.com/stoewer/tempo/pull/1